docs: renew the global resource control rfc (#43389)
pingcap/tidb: docs/design/2022-11-25-global-resource-control.md

CREATE RESOURCE GROUP small_applicatoin RU_PER_SEC=5000;
CREATE RESOURCE GROUP olap_application RU_PER_SEC = 50000 PRIORITY=LOW, BURSTABLE;
/* bind resource group via DCL */
ALTER USER user_name RESOURCE GROUP resouce_group_name;
CREATE USER user_name IDENTIFIED BY 'password' RESOURCE GROUP resource_group_name;

/* bind resource group to the current session */
SET RESOURCE GROUP resouce_group_name

/* bind resource group to sql via optimizer hint */
SELECT /*+ RESOURCE_GROUP(resouce_group_name) */ * from table_name;
```

### Global Quota Control(- Global admission control)

The global admission control can be an independent module that is easy to use and compiled by TiDB. And also, it's possible to be an independent service. Because its local controller part may be installed to the SQL layer first, or it may be moved from the SQL layer to the proxy layer with the evolution. And it also will be related to the billing system in the future. Therefore, it should be as decoupled as possible.
The subsystem can be enabled and disabled. Once enabled, it may bring some performance regression (1%~5% from TopSQL) due to the statistics and control overhead.

Overview

> ![global quota control.png](imgs/global-quota-control.png)

Reviewer: Maybe we can remove this overview

### Request unit

Request unit (short as **RU**) is the uniformed abstract unit of system resource including CPU, IO, etc. Customers explicitly specified the capacity that an application user or tenant required in terms of requesting units (**RU**).

### Request unit Model

Here is to define a cost model relative to the actual cluster resource capacity. Each request will consume the resource capacity. The TiDB layer will cost CPU and memory for different SQL requests, and then SQL requests will transform into the KV requests. These KV requests include different types, requiring the resource of CPU/IO/Memory. All in one, our abstraction unit needs to be mapped to the resources of the cluster, including CPU/IO/Memory, and even the network. For now, we only consider two factors: CPU and IO.

$$RU=\alpha_r + \beta_r * IOBytes + \gamma_r * CpuSecs$$

> The values of the coefficients α,β,γ can be different in different environment, they require datasets from various tests then regress and fit a solution(values) that best maps the actual resource usage.

Reviewer 2: ```suggestion
The values of the coefficients α,β,γ vary in different environment, they require datasets from various tests then regress and fit a solution(values) that best maps the actual resource usage.
```
